### PR TITLE
Adds a new traitor curator firearm to the game.

### PR DESCRIPTION
**Pull request by Thesingleworstcoder**
## About The Pull Request

It's a firearm for traitor curators with a decent technical backstory, and an important difference in affect. It costs more than the standard revolver, and you get a lot less ammo for it (ten shots total), but it makes up for this by being devastatingly effective, putting people into softcrit in two shots instead of the normal three, and killing people with the third shot. It's a weapon for surprise, for ambush, for when you need to make sure that there's one less person between you and your agenda, whatever that may be!

## Why It's Good For The Game

Because antagonists have it hard, and there's nothing worse than getting to the apex of your grand plan only for some random greytide to squeal over the radio your name and location before the third shot lands. It also punishes murderboning, as it's good for at most five people (ten shots, two per person), meaning the antagonist, in turn, needs to pick and choose their targets with caution and diligence.

## Changelog
:cl:
add: Cosmonaut's Revolver
add: Cosmonaut's Re-loader
imageadd: adds the new sprite for the revolver
imagedel: adds the new re-loader sprite
:cl:
Cosmonautrevolver

### CODE REVIEW

**Comment by Thesingleworstcoder** [comment]
tgui/public/tgui-common.bundle.js
not 100% sure what that is, or why it's conflicted.

**Comment by MLGTASTICa** [comment]
Good PR.

**Comment by Gandalf2k15** [comment]
Remove the bundle.js, and also you'll need to put the icons in the modular files D:

**Comment by Thesingleworstcoder** [comment]
> 
> 
> Remove the bundle.js, and also you'll need to put the icons in the modular files D:

I'll get to work!

**Comment by Thesingleworstcoder** [comment]
> 
> 
> Remove the bundle.js, and also you'll need to put the icons in the modular files D:

So I took a look at it, and I don't think it's worth the risk. The conflict isn't doing damage, but deleting the bundle.js is causing more errors than I have fingers and toes. Let's just leave it as-is? I'll still move the sprites though

**Comment by Thesingleworstcoder** [comment]
> 
> 
> > Remove the bundle.js, and also you'll need to put the icons in the modular files D:
> 
> So I took a look at it, and I don't think it's worth the risk. The conflict isn't doing damage, but deleting the bundle.js is causing more errors than I have fingers and toes. Let's just leave it as-is? I'll still move the sprites though

Sprites have been moved

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 1]
```diff
@@ -271,3 +271,11 @@
 		user.emote("scream")
 		user.drop_all_held_items()
 		user.Paralyze(80)
+
+//SKYRAT ADDITION START
+/obj/item/gun/ballistic/revolver/cosmonaut
+	name = "Cosmonaut's Revolver"
+	desc = "You know what? Fuck you. *bullpups your revolver*"
+	icon_state = "cosmonaut"
```
Comment (tf-4) [inline code/modules/projectiles/guns/ballistic/revolver.dm]:
```suggestion
  icon = 'modular_skyrat/modules/modular_weapons/icons/obj/guns/projectile.dmi'
	icon_state = "cosmonaut"
```

**Review by tf-4 — COMMENTED** [review]
Needs to be modularised, shouldn't be too much trouble.

move the icons to the `modular_skyrat/modules/modular_weapons` sub-folders as well as the code itself, and path to the icons appropriately in the code

I like it, though! Good addition.

File: code/modules/projectiles/guns/ballistic/revolver.dm (hunk 1 above)
Comment (tf-4) [reply code/modules/projectiles/guns/ballistic/revolver.dm]: fucking indentation is hard in the web editor but you get the point

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1907,12 +1907,31 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	restricted_roles = list("Curator")
 	limited_stock = 1 //please don't spam deadchat
 
+//SKYRAT EDIT ADDITION BEGIN
+/datum/uplink_item/role_restricted/cosmonauts_revolver
```
[the remark this answers is not in the document]
Comment (Thesingleworstcoder) [reply code/modules/uplink/uplink_items.dm]: The destination you suggested doesn't seem to exist?

File: code/modules/projectiles/guns/ballistic/revolver.dm (hunk 1 above)
Comment (Thesingleworstcoder) [reply code/modules/projectiles/guns/ballistic/revolver.dm:279]: I actually already tried this once, under recommendation from someone on the Discord, but doing it that way made it far harder to locate, made less sense, and just looked like shitcode. Doing it in the normal areas makes it simpler to find, and make more sense.

Comment (Thesingleworstcoder) [reply code/modules/projectiles/guns/ballistic/revolver.dm]: I would happily do this, but I can't find **ANY** of the locations you're referring to.

Comment (Thesingleworstcoder) [reply code/modules/projectiles/guns/ballistic/revolver.dm]: Plus, changing all of these file locations and file paths is going to hurt me on a spiritual level. It works just fine where it is, and it makes logical sense to boot.

**Comment by Thesingleworstcoder** [comment]
I'm not going to modularise it, because the sheer number of file paths that I'd have to change, the amount of shitcode I'd have to impliment, the time it'd take all combine to make me want to keep it as is.
It works, it looks nice, it makes sense, there's no good reason to go through all the effort of moving it now.

**Comment by CameronWoof** [comment]
I don't understand the job-specific theming behind "a gun that shoots really hard". Also, this gun shoots *really* hard.

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (tf-4) [reply code/modules/uplink/uplink_items.dm]: uhh should do?

[the remark this answers is not in the document]
Comment (tf-4) [reply code/modules/uplink/uplink_items.dm]: it's there for me

**Comment by Thesingleworstcoder** [comment]
> 
> 
> I don't understand the job-specific theming behind "a gun that shoots really hard". Also, this gun shoots _really_ hard.

It's not the mechanic that's themed around the job, it's the weapon itself. The curator is by all means quite a weak antagonist, and compared to maxcaps, deathnettle, diseases, or anything else that the other traitors can make with relative ease. The curator needs something strong to make up the deficit in power. And besides, it also has very few shots. It hits hard, but it's tactical, since you only get to shoot it ten times, five if you want to actually kill anyone. On a crew of fourty or higher (normally 80) that's not a huge amount. You can't even produce more ammo for it, since it's a museum piece, befitting the curator.

**Comment by tf-4** [comment]
> 
> 
> I'm not going to modularise it, because the sheer number of file paths that I'd have to change, the amount of shitcode I'd have to impliment, the time it'd take all combine to make me want to keep it as is.
> It works, it looks nice, it makes sense, there's no good reason to go through all the effort of moving it now.

There is no reason not for this to be modularised. All it is is moving a couple of filepaths. If it's not modularised, it could cause issues down the line.
I like the content, it's just non-modular for no reason

**Comment by Thesingleworstcoder** [comment]
> 
> 
> > I'm not going to modularise it, because the sheer number of file paths that I'd have to change, the amount of shitcode I'd have to impliment, the time it'd take all combine to make me want to keep it as is.
> > It works, it looks nice, it makes sense, there's no good reason to go through all the effort of moving it now.
> 
> There is no reason not for this to be modularised. All it is is moving a couple of filepaths. If it's not modularised, it could cause issues down the line.
> I like the content, it's just non-modular for no reason

I couldn't find any of the filepaths you suggested. Not one of them seemed to yield results. Are you suggesting you want me to make these new files?

**Comment by tf-4** [comment]
yes, if they don't already exist

**Comment by Thesingleworstcoder** [comment]
> 
> 
> yes, if they don't already exist

Alright, I didn't think that'd be a good idea, but I can do it if it's that important.

**Comment by Thesingleworstcoder** [comment]
That took WAY too long
It'd help if I had more than like 3 braincells

**Comment by Thesingleworstcoder** [comment]
Modularised!

**Review by KathrinBailey — CHANGES_REQUESTED** [review]
1. First, this needs to be modular, so remove your edits to his grace.
2. Second, don't do the .357x29 thing. It's confusing. Now we have .357 that does 60 damage and .357 that does 70 damage. It's bad enough we have two versions of 9mm. Make some new bullet type or use normal .357
3. Third, what exactly is this? It just seems like some meme gun. Is it a reference to something?
4. Fourth, conflicts.

**Comment by Gandalf2k15** [comment]
Hi, as suggested and requested this must be done modularly, if you wish to continue this you may reopen the PR with the changes, thank you!

**Pull request closed without merging**